### ISSUE TITLE
ES 5.x support?

### ISSUE THREAD

**Issue by cheruvian** [sirensolutions/siren-join#108]
Doesn't look like ES 5.x is supported.

**Comment by turp1twin** [comment]
I think I am going to give that a go, not sure how long it will take given my schedule and it looks like there have been quite a few API changes with 5.0.... Stop me now if this is already in the works.... Anyone? :-)

**Comment by rendel** [comment]
@turp1twin sure feel free to start attacking this issue ;o) I can provide support. I am already working on es 5.x, but for another plugin, so I might have already encountered the API changes that could cause issue. good luck !

**Comment by turp1twin** [comment]
Looks like this is going to be a bit of a bear, lots has changed. For one it looks like all shard level services for Plugins have been removed... Looks like there might be a way around it, but I haven't sorted that out yet. Did you happen to face any of this with the plugin you are porting to 5.0? Cheers!

Jeff

**Comment by rendel** [comment]
Indeed, shard plugin modules has been removed in https://github.com/elastic/elasticsearch/pull/13881. As they recommend, we should be able to use instead an index level service that will listen to shard lifecycle. The index level service will track a version number for multiple shards. It will have to use the `IndexEventListener#afterIndexShardStarted` and `#afterIndexShardClosed`api to keep track on the available shards on the node. And for each shard, it will register a listener using the `IndexingOperationListener` api (as it is now). 
So yes there is a bit of work, as we need to move this shard level service to a node level service, and implement tracking on a per shards basis.
If this looks to complicated, maybe you can just deactivate this service for the moment, and work on the rest of the code.

**Comment by turp1twin** [comment]
Thanks for the feedback, that definitely is the approach I am following. I am knee deep in it right now and will let you know how things progress. If I hit any snags I will let you know... Cheers!


Jeff

**Comment by turp1twin** [comment]
Still working on it, sorry for the delay... Got sidetracked with my day job... Will hopefully have it wrapped up soon...

Jeff

**Comment by tomasfalt** [comment]
Any plan to release this soon?

**Comment by calicartel** [comment]
Hello, is there an ETA on this? Thanks.

**Comment by jccq** [comment]
Best kept secret in industry... It's out as this morning :) see
support.Siren.solutions for kibi 5.2

Announcement later today

**Comment by mayurwish** [comment]
We are currently using only Siren-Join as standalone plugin without Kibi for filter-join, not sure how can  Kibi 5.2.2-beta-1 could let us use filter-join in our Java Project. what jar to include in our build.gradle?

**Comment by jccq** [comment]
please wait for the distribution

*Giovanni Tummarello*   Siren Solutions - CEO
T +353 91704885 | M +353 87 1272938 |
giovanni@siren.solutions | http://siren.solutions/
Unit 13 GTC, Mervue Park, Galway, Ireland
<http://www.twitter.com/jccq?utm_source=WiseStamp&utm_medium=email&utm_term=&utm_content=&utm_campaign=signature>
<http://youtube.com/c/SirenSolutionsChannel?utm_source=WiseStamp&utm_medium=email&utm_term=&utm_content=&utm_campaign=signature>
<https://www.linkedin.com/in/giovannitummarello?utm_source=WiseStamp&utm_medium=email&utm_term=&utm_content=&utm_campaign=signature>

**Comment by libenyang** [comment]
@jccq hi，Is there a plan to be released?

**Comment by AndrewBryukhov** [comment]
waiting for release!

**Comment by frozenspider** [comment]
Waiting for it as well

**Comment by senty** [comment]
Any news guys ? Is it ready for ES 5.x ?

**Comment by juanfmx2** [comment]
The distribution is . . . . . . not here yet. However, in a very tricky way, if you download kibi 5.2.2 
 standard (https://support.siren.solutions/support/solutions/articles/17000049505-kibi-5-2-2-beta-1-community-edition) you can get the plugin for elastic search 5.2.2. Just unzip it and inside the zip there is an elasticsearch 5.2.2 distribution with the plugin installed, or copy the plugin to your own elastic search distribution.

**Comment by rendel** [comment]
@juanfmx2 all

While you can get our new plugin (called vanguard) from the kibi 5.2.2-beta-1 distribution, it is important to understand that this plugin is not open source and not licensed for commercial purposes. In other words, you can only use it for demo and not for commercial application.

We will clarify the licensing terms when we will do an official release.

**Comment by mrdanadams** [comment]
@rendel could you please clarify your comment about licensing above? ^

**Comment by jccq** [comment]
@mrdanadams we will soon have an official release with commercial terms. Likely 1st node is free, next ones require a licence  please write to info@siren.solutions for commercial conversation.

**Comment by mariusbalcytis** [comment]
Will there be open-sourced support for ElasticSearch 5.x? It's been a month - we don't quite know where to go next, as upgrading to 5.x is needed for other reasons and they are beginning to stack together.

**Comment by jccq** [comment]
Hi Marius,

ES 5.x is covered by our new plugin Vanguard. Vanguard is a full rewrite to
support a roadmap which will roll out in the next months featuers like
fully distributed joins and federated queries with other backend system

We are distributing Vanguard as freemium: first data node free, with best
effort support on the forum, the others covered by commercial licence with
enterprise grade support (phone center 24/7 etc available).

Vanguard is available to be licenced standalone or as part of Kibi (
popular choice!)

this week we'll rollout the new website and all the info on this.

thanks

*Giovanni Tummarello*   Siren Solutions - CEO
T +353 91704885 | M +353 87 1272938 |
giovanni@siren.solutions | http://siren.solutions/
Unit 13 GTC, Mervue Park, Galway, Ireland
<https://www.twitter.com/jccq?utm_source=WiseStamp&utm_medium=email&utm_term=&utm_content=&utm_campaign=signature>
<https://youtube.com/c/SirenSolutionsChannel?utm_source=WiseStamp&utm_medium=email&utm_term=&utm_content=&utm_campaign=signature>
<https://www.linkedin.com/in/giovannitummarello?utm_source=WiseStamp&utm_medium=email&utm_term=&utm_content=&utm_campaign=signature>
